index.html

```diff
@@ -159,141 +156,401 @@ <h2>
+            root=], then:
+              <ol>
+                <li>If [=options=] are equivalent to the current [=pointer-lock
+                options=]
+                  <ol>
+                    <li>Set [=pointer-lock target=] to [=this=]
+                    </li>
+                    <li>[=Queue an element task=] on the [=user interaction
+                    task source=], given this, to perform the following steps:
+                      <ol>
+                        <li>[=Fire an event=] named {{pointerlockchange}} at
+                        [=this=]'s [=Node/node document=].
+                        </li>
+                        <li>[=Resolve=] the <var>promise</var>.
+                        </li>
+                      </ol>
+                    </li>
+                  </ol>
+                </li>
+                <li>If [=options=] are not equivalent to the user agent's
+                current [=pointer-lock options=] and [=options=] are not
+                supported:
+                  <ol>
+                    <li>[=Queue an element task=] on the [=user interaction
+                    task source=], given this, to perform the following steps:
+                      <ol type='i'>
+                        <li>[=Reject=] <var>promise</var> with a
+                        "{{NotSupportedError}}" {{DOMException}}.
+                        </li>
+                      </ol>
+                    </li>
+                    <li>Return <var>promise</var>.
+                    </li>
+                  </ol>
+                </li>
+                <li>If [=options=] are not equivalent to the current
+                [=pointer-lock options=] and [=options=] are supported then
+                perform the following steps
+                  <ol>
+                    <li>Update the user agent's [=pointer-lock target=] to this
+                    element.
+                    </li>
+                    <li>Update the user agent's [=pointer-lock options=] to
+                    [=options=]
+                    </li>
+                    <li>[=Queue an element task=] on the [=user interaction
+                    task source=], given this, to perform the following steps:
+                      <ol type='i'>
+                        <li>[=Fire an event=] named {{pointerlockchange}} at
+                        [=this=]'s [=Node/node document=].
+                        </li>
+                        <li>The <var>promise</var> must be resolved.
+                        </li>
+                      </ol>
+                    </li>
+                    <li>Return <var>promise</var>.
+                    </li>
+                  </ol>
+                </li>
+              </ol>
+            </li>
+            <li>If the request was not started from an [=engagement gesture=]
+            and the {{Document}} has not previously released a successful Pointer
+            Lock with {{Document/exitPointerLock()}}:
```

Comment:
Wait!!!! We should fix this so that it requires "transient activation" or "sticky activation"  based on the lock already being applied. The "engagement gesture" is not a thing (or should not be a thing defined/used in this spec).

Reply:
Also, this should have been checked way earlier... probably the first thing this method should check is if it has transient activation or not.... and return a promise rejected. Also, if you are passing the option, maybe we don't need to fire the event? 

Having both models doesn't seem ideal. Sending the parameter should opt you into the promise model.